xds/src/main/java/io/grpc/xds/orca/OrcaMetricReportingServerInterceptor.java

```diff
@@ -46,27 +47,40 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/9127")
 public final class OrcaMetricReportingServerInterceptor implements ServerInterceptor {
 
-  private static final OrcaMetricReportingServerInterceptor INSTANCE =
-      new OrcaMetricReportingServerInterceptor();
+  private static volatile OrcaMetricReportingServerInterceptor instance;
 
   @VisibleForTesting
   static final Metadata.Key<OrcaLoadReport> ORCA_ENDPOINT_LOAD_METRICS_KEY =
       Metadata.Key.of(
           "endpoint-load-metrics-bin",
           ProtoUtils.metadataMarshaller(OrcaLoadReport.getDefaultInstance()));
 
-  private volatile MetricRecorder metricRecorder;
+  private final MetricRecorder metricRecorder;
 
   @VisibleForTesting
-  OrcaMetricReportingServerInterceptor() {
+  OrcaMetricReportingServerInterceptor(MetricRecorder metricRecorder) {
+    this.metricRecorder = metricRecorder;
+    OrcaMetricReportingServerInterceptor.instance = this;
```

Comment:
No, we shouldn't want to leak the metric recorder into a global. Remove this line. If a user calls the old API, they get the old behavior.

Reply:
Done.